test/TestConfig/scripts/testKitGen/makeGenTool/mkgen.pl

```diff
@@ -614,7 +593,17 @@ sub writeTargets {
 			print $fhOut "$indent\@echo \"===============================================\" | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";
 			print $fhOut "$indent\@echo \"Running test \$\@ ...\" | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";
 			print $fhOut "$indent\@echo \"===============================================\" | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";
-			print $fhOut "$indent\@perl \'-MTime::HiRes=gettimeofday\' -e \'print \"$name Start Time: \" . localtime() . \" Epoch Time (ms): \" . int (gettimeofday * 1000) . \"\\n\"\' | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";
+			print $fhOut "$indent\@perl \'-MTime::HiRes=gettimeofday\' -e \'print \"$name Start Time: \" . localtime() . \" Epoch Time (ms): \" . int (gettimeofday * 1000) . \"\\n\"\' | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";			
+			if (defined $test->{'disabled'}) {
+				#This line is also the key words to match runningDisabled
+				print $fhOut "$indent\@echo \"Test is disabled due to:\"  | tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";
+				my @disabledReasons = split("[\t\n]", $test->{'disabled'});
+				foreach my $dReason (@disabledReasons) {
+					if ($dReason ne "") {
+						print $fhOut "$indent\@echo \"$dReason\"\n";
```

Comment:
Add `| tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";` in the the print out.